core-it-suite/src/test/java/org/apache/maven/it/ItUtils.java

```diff
@@ -56,4 +63,24 @@ public static String calcHash(File file, String algo) throws Exception {
 
         return hash.toString();
     }
+
+    public static String getLogContent(Verifier verifier) throws IOException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        Files.copy(Paths.get(verifier.getBasedir(), verifier.getLogFileName()), baos);
+        return baos.toString();
+    }
+
+    public static void setUserHome(Verifier verifier, File file) {
+        setUserHome(verifier, file.toPath());
+    }
+
+    public static void setUserHome(Verifier verifier, Path home) {
+        // NOTE: We set the user.home directory instead of say settings.security to reflect Maven's normal behavior
+        String path = home.toAbsolutePath().toString();
+        if (path.indexOf(' ') < 0) {
+            verifier.setEnvironmentVariable("MAVEN_OPTS", "-Duser.home=" + path);
+        } else {
+            verifier.setEnvironmentVariable("MAVEN_OPTS", "\"-Duser.home=" + path + "\"");
```

Comment:
We also need here 
```
        verifier.setForkJvm(true);
```
environment variable can be set only in fork mode of verifier

Reply:
It's not trivial to get the forkJvm value, therefore i've only added a comment to warn future developers about it. I hope you're ok with it.